packages/features/ee/round-robin/roundRobinReassignment.ts

```diff
@@ -0,0 +1,517 @@
+        name: previousRRHost.name || "",
+        email: previousRRHost.email,
+        language: {
+          locale: previousRRHost.locale || "en",
+          translate: previousRRHostT,
+        },
+        timeZone: previousRRHost.timeZone,
+        timeFormat: getTimeFormatStringFromUserTimeFormat(previousRRHost.timeFormat),
+      };
+    } else if (cancelledRRHostEvt.team) {
+      // Filter out the new RR host from attendees and add the old RR host
+      const newMembersArray = cancelledRRHostEvt.team?.members || [];
+      cancelledRRHostEvt.team.members = newMembersArray.filter(
+        (member) => member.email !== reassignedRRHost.email
+      );
+      cancelledRRHostEvt.team.members.unshift({
+        id: previousRRHost.id,
+        email: previousRRHost.email,
+        name: previousRRHost.name || "",
+        timeZone: previousRRHost.timeZone,
+        language: { translate: previousRRHostT, locale: previousRRHost.locale || "en" },
+      });
+    }
+
+    await sendRoundRobinCancelledEmails(cancelledRRHostEvt, [
+      {
+        ...previousRRHost,
+        name: previousRRHost.name || "",
+        username: previousRRHost.username || "",
+        timeFormat: getTimeFormatStringFromUserTimeFormat(previousRRHost.timeFormat),
+        language: { translate: previousRRHostT, locale: previousRRHost.locale || "en" },
+      },
+    ]);
+  }
+
+  // Handle changing workflows with organizer
+  if (hasOrganizerChanged) {
+    const workflowReminders = await prisma.workflowReminder.findMany({
+      where: {
+        bookingUid: booking.uid,
+        method: WorkflowMethods.EMAIL,
+        workflowStep: {
+          action: WorkflowActions.EMAIL_HOST,
+          workflow: {
+            trigger: {
+              in: [
+                WorkflowTriggerEvents.BEFORE_EVENT,
+                WorkflowTriggerEvents.NEW_EVENT,
+                WorkflowTriggerEvents.AFTER_EVENT,
+              ],
+            },
+          },
+        },
+      },
+      include: {
+        workflowStep: {
+          include: {
+            workflow: true,
+          },
+        },
+      },
+    });
+
+    for (const workflowReminder of workflowReminders) {
+      const workflowStep = workflowReminder?.workflowStep;
+      const workflow = workflowStep?.workflow;
+
+      if (workflowStep && workflow) {
+        await scheduleEmailReminder({
+          evt: {
+            ...evt,
+            eventType,
+          },
+          action: WorkflowActions.EMAIL_HOST,
+          triggerEvent: workflow.trigger,
+          timeSpan: {
+            time: workflow.time,
+            timeUnit: workflow.timeUnit,
+          },
+          sendTo: reassignedRRHost.email,
+          template: workflowStep.template,
+        });
+      }
+
+      await deleteScheduledEmailReminder(workflowReminder.id, workflowReminder.referenceId);
+    }
+    // Send new event workflows to new organizer
+    const newEventWorkflows = await prisma.workflow.findMany({
```

Comment:
Can we maybe use `getAllWorkflowsFromEventType` or `getAllWorkflows` here? 

One thing I see missing in this query is the workflows that are 'active on all event types or teams' enabled.

Reply:
What are your thoughts here. I only want workflows where the trigger is `NEW_EVENT` and only select the steps with the action `EMAIL_HOST`. I'm thinking if I got all the workflows I'd have to iterate through the workflows array and inside of them iterate over the steps array for O(m * n). Instead I would rather the database do that filtering.

I will add in the `OR` `{isActiveOnAll: true}`

Reply:
Yes, you are right, we should only query the necessary workflows 👍